core/math/SCsub

```diff
@@ -2,4 +2,33 @@
 
 Import('env')
 
-env.add_source_files(env.core_sources, "*.cpp")
+env_math = env.Clone() # Maybe make one specific for crypto?
+
+is_builtin = env["builtin_mbedtls"]
+has_module = env["module_mbedtls_enabled"]
+
+if is_builtin or not has_module:
+    # Use our headers for builtin or if the module is not going to be compiled.
+    # We decided not to depend on system mbedtls just for these few files that can
+    # be easily extracted.
+    env_math.Prepend(CPPPATH=["#thirdparty/mbedtls/include"])
+
+# MbedTLS core functions (CryptoCore)
+if not has_module:
```

Comment:
This should also be `is_builtin or not has_module`, no? If we're `not is_builtin and has_module`, then we should use the system headers since we already link `-lmbedtls`.

Reply:
Added a small comment to make this clearer:
```
# If the mbedtls module is compiled we don't need to add the .c files with our
# custom config since they will be built by the module itself.
# Only if the module is not enabled, we must compile here the required sources
# to make a "light" build with only the necessary mbedtls files.
```
Am I missing the point?

Reply:
You're right, I was confused again :P

Indeed if we don't build the module, we won't link the system mbedtls, so we should use the builtin source in any case.